embassy-rp/src/uart/mod.rs

```diff
@@ -187,9 +188,21 @@ impl<'d, T: Instance, M: Mode> UartTx<'d, T, M> {
     pub fn blocking_flush(&mut self) -> Result<(), Error> {
         let r = T::regs();
         while !r.uartfr().read().txfe() {}
+        while self.busy() {}
         Ok(())
     }
 
+    /// Flush the UART TX, doing a yield-spinloop until done
+    pub async fn yielding_flush(&self) {
```

Comment:
this should be `pub async fn flush()` for consistency (other async fns aren't prefixed like `async_` or `yielding_` , only blocking is prefixed with `blocking_`.)